plugins/shim/plugin.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
+	smith_plugin "github.com/atlassian/smith/pkg/plugin"
+	sc_v1b1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Process(resource smith_v1.Resource, dependencies map[smith_v1.ResourceName]smith_plugin.Dependency) (smith_plugin.ProcessResult, error) {
+	// This is dumb, and just spits out a random service instance
+	name := resource.PluginSpec.Name
+	svcInstance := &sc_v1b1.ServiceInstance{
```

Comment:
So this is the place where we'll do the gRPC call or the REST call, or fork/pipe magic (whatever we agree to), right?

Reply:
if we go the RPC path, I think we should get rid of Go plugins completely, and just stick with the RPC mechanism we agree to. Reasoning for that:
- Go plugins are not mature in general
- It's an unnecessary layer of abtraction, since we unlikely to have other Go plugin implementations in the near future
- with Go plugins it will be hard to init plugin with config (list of supported plugins/endpoints, for example), and cache config between the calls (if we need it)
- it will make Smith build easier, and plugin builds will be a completely separate process. All that we'll need is a config for Smith, which can be either passed as command line parameters, or through ConfigMap stored in Kubernetes.